### ISSUE TITLE
Upgrade to support sonarqube 8.0

### ISSUE THREAD

**Issue by thib3113** [mc1arke/sonarqube-community-branch-plugin#38]
**Describe the bug**
For some reason, we need to upgrade to 8.0 ( sonarqube db is destroy, and 8.0 repair it ... ). 
Did you plan to adapt the plugin for 8.0 ?

Here is the error from sonarqube :
```
java.lang.IllegalStateException: Fail to load plugin Community Branch Plugin [communityBranchPlugin]
	at org.sonar.server.plugins.ServerExtensionInstaller.installExtensions(ServerExtensionInstaller.java:88)
	at org.sonar.ce.container.ComputeEngineContainerImpl.startLevel4(ComputeEngineContainerImpl.java:230)
	at org.sonar.ce.container.ComputeEngineContainerImpl.start(ComputeEngineContainerImpl.java:196)
	at org.sonar.ce.ComputeEngineImpl.startup(ComputeEngineImpl.java:45)
	at org.sonar.ce.app.CeServer$CeMainThread.attemptStartup(CeServer.java:163)
	at org.sonar.ce.app.CeServer$CeMainThread.run(CeServer.java:141)
Caused by: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "getClassLoader")
	at java.base/java.security.AccessControlContext.checkPermission(Unknown Source)
	at java.base/java.security.AccessController.checkPermission(Unknown Source)
	at java.base/java.lang.SecurityManager.checkPermission(Unknown Source)
	at java.base/java.lang.ClassLoader.checkClassLoaderPermission(Unknown Source)
	at java.base/java.lang.Class.getClassLoader(Unknown Source)
	at com.github.mc1arke.sonarqube.plugin.classloader.ClassReferenceElevatedClassLoaderFactory.createClassLoader(ClassReferenceElevatedClassLoaderFactory.java:47)
	at com.github.mc1arke.sonarqube.plugin.CommunityBranchPluginBootstrap.define(CommunityBranchPluginBootstrap.java:60)
	at org.sonar.server.plugins.ServerExtensionInstaller.installExtensions(ServerExtensionInstaller.java:78)
	... 5 common frames omitted
```

**To Reproduce**
1. Install sonarqube 8.0
2. Add this plugin in plugin folder
3. start sonarqube

**Expected behavior**
work

**Software Versions**
 - SonarQube Version: 8.0-beta
 - Plugin Version: 1.2.0-snapshot

**Additional context**
First, I'm not a java developper . 

I try this : 
Go in `build.gradle`,
change `sonarqubeVersion` to `8.0`
start `gradlew check`
return this : 
```
C:\Users\thib3\repos\sonarqube-community-branch-plugin\src\main\java\com\github\mc1arke\sonarqube\plugin\scanner\CommunityProjectBranchesLoader.java:28: error: cannot find symbol
import org.sonar.scanner.util.ScannerUtils;
```
( this is not the full stacktrace ) .
If I go in the file CommunityProjectBranchesLoader.java, I see intellij idea can't resolve all import from sonarqube ( and from google ) ... ( and idea can't see class inside the .java file for 8.0, but can for 7.8 ) . 

I don't know how to help, but If you need help, you can ask :) .

**Comment by mc1arke** [comment]
SonarSource have made changes to specifically block the mechanism this plugin used to access the relevant Core classes at runtime by [implementing a Security Manager](https://github.com/SonarSource/sonarqube/commit/10a1454ce0e0b97f00e2792942f1e2131bc877db). Alongside this they've made internal API changes that prevent the plugin compiling against SonarQube 8. I'll need to take a look at what options we have for overcoming this.

**Comment by mc1arke** [comment]
@thib3113 @simkin could you build and run #41 and confirm whether it works for you? I've successfully run it locally against SonarQube 7.8 and 8.0 both with branch and without branch parameters. Be aware of the slight change in installation instructions included in the README file. If it works OK then I'll get it merged and released.

**Comment by simkin** [comment]
I never compiled a plugin before, would you be so kind to link me to some instructions? Happy to compile and test when I learned how to :)

**Comment by mc1arke** [comment]
You'll need to have a JDK installed locally. It's then a case of running:
```
git clone https://github.com/mc1arke/sonarqube-community-branch-plugin.git
cd sonarqube-community-branch-plugin
git checkout sq-8_0-support
./gradlew build
```
This will create the plugin Jar in the`build/libs/` directory of the project which is the file you need to copy so `$SONARQUBE_HOME/libs/common` and `$SONARQUBE_HOME/extensions/plugins`.

**Comment by thib3113** [comment]
I've tried to install it yesterday, and "it's work", sonarqube doesn't crash ... But I have another big, and can't see the projects (not linked to this plugin I think), so, I can't help more :/ (I 'll debug my problem, and check the branch when I can see the projects)

**Comment by dpandolfo** [comment]
Hi @mc1arke , 

I tested the 1.2.0-SNAPSHOT plugin with Sonarqube 8.0 and it worked normally.

The only problem I noticed was when the inspection is first performed, the project is not created automatically. So I need to create the project before it runs, just that works.

Could this be related to issue #1 ?


Here is the error from sonarqube :

`18:26:28.103 INFO: Load project settings for component key: 'investimentos-poupanca.poupancatradicional-debito-api'
18:26:28.115 DEBUG: GET 404 http://####/api/settings/values.protobuf?component=investimentos-poupanca.poupancatradicional-debito-api | time=12ms
18:26:28.137 INFO: Load project branches
18:26:28.145 DEBUG: GET 404 http://####/api/project_branches/list?project=investimentos-poupanca.poupancatradicional-debito-api | time=8ms
18:26:28.161 INFO: ------------------------------------------------------------------------
18:26:28.162 INFO: EXECUTION FAILURE
18:26:28.162 INFO: ------------------------------------------------------------------------
18:26:28.162 INFO: Total time: 7.594s
18:26:28.197 INFO: Final Memory: 7M/30M
18:26:28.197 INFO: ------------------------------------------------------------------------
18:26:28.197 ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Unable to load component class org.sonar.scanner.scan.filesystem.InputComponentStore
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:65)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:632)
        at org.picocontainer.parameters.BasicComponentParameter$1.resolveInstance(BasicComponentParameter.java:118)
        at org.picocontainer.parameters.ComponentParameter$1.resolveInstance(ComponentParameter.java:136)
        at org.picocontainer.injectors.SingleMemberInjector.getParameter(SingleMemberInjector.java:78)
        at org.picocontainer.injectors.ConstructorInjector$CtorAndAdapters.getParameterArguments(ConstructorInjector.java:309)
        at org.picocontainer.injectors.ConstructorInjector$1.run(ConstructorInjector.java:335)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.ConstructorInjector.getComponentInstance(ConstructorInjector.java:364)
        at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.getComponentInstance(AbstractInjectionFactory.java:56)
        at org.picocontainer.behaviors.AbstractBehavior.getComponentInstance(AbstractBehavior.java:64)
        at org.picocontainer.behaviors.Stored.getComponentInstance(Stored.java:91)
        at org.picocontainer.DefaultPicoContainer.instantiateComponentAsIsStartable(DefaultPicoContainer.java:1034)
        at org.picocontainer.DefaultPicoContainer.addAdapterIfStartable(DefaultPicoContainer.java:1026)
        at org.picocontainer.DefaultPicoContainer.startAdapters(DefaultPicoContainer.java:1003)
        at org.picocontainer.DefaultPicoContainer.start(DefaultPicoContainer.java:767)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
        at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
        at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
        at com.sun.proxy.$Proxy0.execute(Unknown Source)
        at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
        at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
        at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
        at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
        at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: Unable to load component interface org.sonar.scanner.scan.branch.BranchConfiguration
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:65)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:632)
        at org.picocontainer.parameters.BasicComponentParameter$1.resolveInstance(BasicComponentParameter.java:118)
        at org.picocontainer.parameters.ComponentParameter$1.resolveInstance(ComponentParameter.java:136)
        at org.picocontainer.injectors.SingleMemberInjector.getParameter(SingleMemberInjector.java:78)
        at org.picocontainer.injectors.ConstructorInjector$CtorAndAdapters.getParameterArguments(ConstructorInjector.java:309)
        at org.picocontainer.injectors.ConstructorInjector$1.run(ConstructorInjector.java:335)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.ConstructorInjector.getComponentInstance(ConstructorInjector.java:364)
        at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.getComponentInstance(AbstractInjectionFactory.java:56)
        at org.picocontainer.behaviors.AbstractBehavior.getComponentInstance(AbstractBehavior.java:64)
        at org.picocontainer.behaviors.Stored.getComponentInstance(Stored.java:91)
        at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:63)
        ... 34 more
Caused by: java.lang.IllegalStateException: Unable to load component class org.sonar.scanner.scan.branch.ProjectBranches
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:65)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:632)
        at org.picocontainer.parameters.BasicComponentParameter$1.resolveInstance(BasicComponentParameter.java:118)
        at org.picocontainer.parameters.ComponentParameter$1.resolveInstance(ComponentParameter.java:136)
        at org.picocontainer.injectors.SingleMemberInjector.getParameter(SingleMemberInjector.java:78)
        at org.picocontainer.injectors.SingleMemberInjector.getMemberArguments(SingleMemberInjector.java:61)
        at org.picocontainer.injectors.MethodInjector.getMemberArguments(MethodInjector.java:100)
        at org.picocontainer.injectors.MethodInjector$2.run(MethodInjector.java:112)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.MethodInjector.decorateComponentInstance(MethodInjector.java:120)
        at org.picocontainer.injectors.CompositeInjector.decorateComponentInstance(CompositeInjector.java:58)
        at org.picocontainer.injectors.Reinjector.reinject(Reinjector.java:142)
        at org.picocontainer.injectors.ProviderAdapter.getComponentInstance(ProviderAdapter.java:96)
        at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:63)
        ... 48 more
Caused by: java.lang.IllegalStateException: Could not execute ScannerWsClient
        at com.github.mc1arke.sonarqube.plugin.scanner.ScannerWsClientWrapper.call(ScannerWsClientWrapper.java:42)
        at com.github.mc1arke.sonarqube.plugin.scanner.CommunityProjectBranchesLoader.load(CommunityProjectBranchesLoader.java:66)
        at org.sonar.scanner.scan.branch.ProjectBranchesProvider.provide(ProjectBranchesProvider.java:48)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at org.picocontainer.injectors.MethodInjector.invokeMethod(MethodInjector.java:129)
        at org.picocontainer.injectors.MethodInjector.access$000(MethodInjector.java:39)
        at org.picocontainer.injectors.MethodInjector$2.run(MethodInjector.java:113)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.MethodInjector.decorateComponentInstance(MethodInjector.java:120)
        at org.picocontainer.injectors.CompositeInjector.decorateComponentInstance(CompositeInjector.java:58)
        at org.picocontainer.injectors.Reinjector.reinject(Reinjector.java:142)
        at org.picocontainer.injectors.ProviderAdapter.getComponentInstance(ProviderAdapter.java:96)
        at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:63)
        ... 63 more
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.github.mc1arke.sonarqube.plugin.scanner.ScannerWsClientWrapper.call(ScannerWsClientWrapper.java:40)
        ... 80 more
Caused by: org.sonarqube.ws.client.HttpException: Error 404 on http://####/api/project_branches/list?project=investimentos-poupanca.poupancatradicional-debito-api : {"errors":[{"msg":"Component key 'investimentos-poupanca.poupancatradicional-debito-api' not found"}]}
        at org.sonarqube.ws.client.BaseResponse.failIfNotSuccessful(BaseResponse.java:36)
        at org.sonar.scanner.bootstrap.DefaultScannerWsClient.failIfUnauthorized(DefaultScannerWsClient.java:112)
        at org.sonar.scanner.bootstrap.DefaultScannerWsClient.call(DefaultScannerWsClient.java:75)
        ... 85 more`

**Comment by mc1arke** [comment]
@dpandolfo Thanks for the report, and you're correct that this is a regression of #1. I've therefore corrected the tests that should have protected this feature and fixed the feature in question. Could you test the updated version please?

**Comment by dpandolfo** [comment]
Hi @mc1arke, very good! I performed new tests and the plug-in worked as expected.

Thanks!  :smiley:

**Comment by thib3113** [comment]
Hello @mc1arke , sorry for the long delay, I'm trying to setup a new sonarqube with this update ... 

And I see in this comment : https://github.com/mc1arke/sonarqube-community-branch-plugin/issues/38#issuecomment-546232843, you say `$SONARQUBE_HOME/libs/common`, but in the readme, you say `lib/common`, it's the `lib` forlder or the `libs` folder ? ( I'm using docker, I can't do a `ls`, and I think you will need to update the comment/readmeme, to avoid futur confusion ) .

**Comment by simkin** [comment]
> You'll need to have a JDK installed locally. It's then a case of running:
> 
> ```
> git clone https://github.com/mc1arke/sonarqube-community-branch-plugin.git
> cd sonarqube-community-branch-plugin
> git checkout sq-8_0-support
> ./gradlew build
> ```
> 
> This will create the plugin Jar in the`build/libs/` directory of the project which is the file you need to copy so `$SONARQUBE_HOME/libs/common` and `$SONARQUBE_HOME/extensions/plugins`.

All works as it should, tested thoroughly. Thanks for the fix!

**Comment by thib3113** [comment]
`but an obvious one (there is only one lib* dir)` : not if you use docker 😉 . I can't see the internal folders, just write a mapping, and if the folder doesn't exist, docker will just create an empty folder 🙂 .
( and can't go in `$SONARQUBE_HOME` to do an `ls`, because the container crash fastly, so too fast to use shell inside )...


So, on my side, when I launch the sonar-scanner (cli), I get this on sonarqube : 
```
2019.11.09 16:11:41 ERROR web[AW5Q7T24UZpJh26CAAA7][o.s.s.w.WebServiceEngine] Fail to process request http://sonar.monsite.com/api/ce/submit?projectKey=my-API&projectName=my%20Api&characteristic=branch%3Ddev&characteristic=branchType%3DSHORT

java.lang.IllegalStateException: Current edition does not support branch feature
	at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
	at org.sonar.server.ce.queue.BranchSupport.createComponentKey(BranchSupport.java:63)
	at org.sonar.server.ce.queue.ReportSubmitter.submit(ReportSubmitter.java:83)
	at org.sonar.server.ce.ws.SubmitAction.handle(SubmitAction.java:113)
	at org.sonar.server.ws.WebServiceEngine.execute(WebServiceEngine.java:110)
	at org.sonar.server.platform.web.WebServiceFilter.doFilter(WebServiceFilter.java:92)
	at org.sonar.server.platform.web.MasterServletFilter$GodFilterChain.doFilter(MasterServletFilter.java:138)
	at org.sonar.server.platform.web.MasterServletFilter.doFilter(MasterServletFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.sonar.server.platform.web.UserSessionFilter.doFilter(UserSessionFilter.java:88)
	at org.sonar.server.platform.web.UserSessionFilter.doFilter(UserSessionFilter.java:72)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.sonar.server.platform.web.CacheControlFilter.doFilter(CacheControlFilter.java:76)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.sonar.server.platform.web.SecurityServletFilter.doHttpFilter(SecurityServletFilter.java:76)
	at org.sonar.server.platform.web.SecurityServletFilter.doFilter(SecurityServletFilter.java:48)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.sonar.server.platform.web.RedirectFilter.doFilter(RedirectFilter.java:58)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.sonar.server.platform.web.RequestIdFilter.doFilter(RequestIdFilter.java:66)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.sonar.server.platform.web.RootFilter.doFilter(RootFilter.java:62)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.filters.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:109)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:256)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:808)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Unknown Source)
```
And the scanner failed without any errors ... just some warnings, I think they are not linked, but in case : 

Seems to be a bug on the sonar js scanner
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by net.sf.cglib.core.ReflectUtils$1 (file:/home/gitlab-runner/.sonar/cache/866bb1adbf016ea515620f1aaa15ec53/sonar-javascript-plugin.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of net.sf.cglib.core.ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
``` 

```
WARN: File '/home/gitlab-runner/builds/7NUff9af/0/myproject/API/src/objects/rdv/rdvTokenManager.ts' was detected as changed but without having changed lines
WARN: File '/home/gitlab-runner/builds/7NUff9af/0/myproject/API/src/d/object/DSStaffRole/IDSStaffRoleParams.ts' was detected as changed but without having changed lines
WARN: File '/home/gitlab-runner/builds/7NUff9af/0/myproject/API/tests/object/List.test.ts' was detected as changed but without having changed lines
WARN: File '/home/gitlab-runner/builds/7NUff9af/0/myproject/API/src/d/object/DSIndividualRole/IDSIndividualRoleParams.ts' was detected as changed but without having changed lines
```

**Comment by nixel2007** [comment]
> I can't see the internal folders, just write a mapping,

```sh
docker exec -it containerName bash
```

or

```sh
docker run -it --rm imageName bash
```

**Comment by thib3113** [comment]
@nixel2007 not so easy too ... 
I didn't have an ssh access to this machine ... So to do this, I need to go at work, but it's closes before tuesday, so not possible .

Other things, I can create a new vm (or dual boot), install docker, and try this, but too long, and not a good idea too 

After this, I need to convert my stack file to a command line, If I want to use the exact same environment ...

And, using "bash" is not a good idea, if you want a generic command, use /bin/sh (some alpine images doesn't contain bash ... ), and if the image is directly crashing, you doesn't have the time to do the `ls` after starting the shell .

Finally, I things it's better to correct this little typo, than answering peoples reading this typo, and do mapping with the typo ... 

So, can we stop talking about this out of scope problem ? 

---

In other point, 
I see docker image use two env for home `$SONARQUBE_HOME` and `$SONARQUBE_PUBLIC_HOME`, 
this two folders contains a `lib/comon` folder ... 

with mapping `$SONARQUBE_HOME/lib/common` contain only this jar file (this plugin builded), and `$SONARQUBE_PUBLIC_HOME/lib/common` seems to contain lot of jar files ...
 
So, did I need to use `$SONARQUBE_HOME`, or add this jar with the lot of jar files (maybe this is why I get the error ?) ?

**Comment by thib3113** [comment]
So I use `$SONARQUBE_PUBLIC_HOME/lib/common`, and I rebuild the plugin (seems I've done some weird things, so I restart with a clean checkout) . And it's work for me, thank you 🙂

**Comment by mc1arke** [comment]
To confirm - given the above conversation - the plugin needs copied to `lib/common` in the SonarQube base directory as there is no `libs` directory. I've fixed the typo in my original instruction on how to build an deploy the project, however the README file on this branch remains correct.

Given the only reports I've had from people trying this latest version, I'll get it merged and released this week.

**Comment by nk111** [comment]
I builded the 8.0 snapshot and installed it on our 8.0 test installation. The plugin is listed under "installed" but otherwise there is nothing changed in the UI. When we try to analyze a feature branch we get an unspecific error on the client side and this output in the web.log on the server side:
`2019.11.15 12:10:46 ERROR web[xxx][o.s.s.w.WebServiceEngine] Fail to process request http://server_url/api/ce/submit?projectKey=xxx$
java.lang.IllegalStateException: Current edition does not support branch feature`
Anything idea what might be wrong?

**Comment by simkin** [comment]
Did you copy the plugin to both locations?

**Comment by thib3113** [comment]
@nk111 did add the jar in `/lib/common` ?

**Comment by mc1arke** [comment]
Released as part of 1.2.0
